tests/_helpers.py

```
@@ -0,0 +1,60 @@
+def confirm_failure(parser, test, msg):
+    """Confirm that parsing the test string with the given parser fails or only
+    partially matches."""
+
+    try:
+        pairs = parser.parse_string(test).as_list()
+
+    except ParseException:
+        with pytest.raises(ParseException):
+            parser.parse_string(test)
+
+    except Exception as e:  # pragma: no cover
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)